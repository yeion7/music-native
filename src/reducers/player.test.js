import React from "react";
import "react-native";

import {
  setSong,
  setAlbum,
  changeExpand,
  showPlayer,
  setIndex,
  loadSong,
  setDuration,
  setPosition,
  setPlaying,
  setPlayback
} from "./player";
import {
  onPressSong,
  showPlayer as onShowPlayer,
  onPressAlbum,
  expandPlayer,
  setIndex as onSetIndex,
  setPlayback as onSetPlayback,
  isLoadingSong,
  setPosition as onSetPosition,
  setDuration as onSetDuration,
  setPlaying as onSetPlaying
} from "../actions/player";

describe("search reducer test ", () => {
  it("handles CHANGE_PLAYING_SONG action", () => {
    expect(setSong(null, onPressSong({}))).toMatchSnapshot();
  });
  it("handles CHANGE_SELECTED_PLAYLISTS action", () => {
    expect(setAlbum(null, onPressAlbum({}))).toMatchSnapshot();
  });
  it("handles CHANGE_EXPAND_PLAYER action", () => {
    expect(changeExpand(null, expandPlayer(false))).toMatchSnapshot();
  });
  it("handles CHANGE_PLAYER_STATUS action", () => {
    expect(showPlayer(null, onShowPlayer(false))).toMatchSnapshot();
  });
  it("handles SET_INDEX_SONG action", () => {
    expect(setIndex(null, onSetIndex(1))).toMatchSnapshot();
  });
  it("handles LOADING_PLAYBACK_SONG action", () => {
    expect(loadSong(null, isLoadingSong(true))).toMatchSnapshot();
  });
  it("handles SET_PLAYBACK_DURATION action", () => {
    expect(setDuration(null, onSetDuration(1000))).toMatchSnapshot();
  });
  it("handles UPDATE_PLAYBACK_POSITION action", () => {
    expect(setPosition(null, onSetPosition(1000))).toMatchSnapshot();
  });
  it("handles CHANGE_STATE_SONG action", () => {
    expect(setPlaying(null, onSetPlaying(true))).toMatchSnapshot();
  });
  it("handles SET_PLAYBACK_SONG action", () => {
    expect(setPlayback(null, onSetPlayback({}))).toMatchSnapshot();
  });
});
