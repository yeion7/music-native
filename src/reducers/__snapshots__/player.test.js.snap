// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search reducer test  handles CHANGE_EXPAND_PLAYER action 1`] = `true`;

exports[`search reducer test  handles CHANGE_PLAYER_STATUS action 1`] = `false`;

exports[`search reducer test  handles CHANGE_PLAYING_SONG action 1`] = `Object {}`;

exports[`search reducer test  handles CHANGE_SELECTED_PLAYLISTS action 1`] = `Object {}`;

exports[`search reducer test  handles CHANGE_STATE_SONG action 1`] = `true`;

exports[`search reducer test  handles LOADING_PLAYBACK_SONG action 1`] = `true`;

exports[`search reducer test  handles SET_INDEX_SONG action 1`] = `1`;

exports[`search reducer test  handles SET_PLAYBACK_DURATION action 1`] = `1000`;

exports[`search reducer test  handles SET_PLAYBACK_SONG action 1`] = `Object {}`;

exports[`search reducer test  handles UPDATE_PLAYBACK_POSITION action 1`] = `1000`;
